pkg/manifests/types.go

```diff
@@ -122,6 +122,8 @@ type K8sPrometheusAdapter struct {
 }
 
 // The `MetricsServerConfig` resource defines settings for the MetricsServer component.
+// This resource is only considered valid in a TechPreview cluster
+// with TechPreviewNoUpgrade FeatureSet enabled; otherwise, it is ignored.
```

Comment:
taking inspiration from https://github.com/openshift/cluster-monitoring-operator/pull/2161

cc @bburt-rh 

```suggestion
// Note that this setting only applies when the MetricsServer feature gate is enabled.
```